### PR TITLE
feat: transactional writes from the callee point of view

### PR DESCRIPTION
**Pull request by ppedziwiatr**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -16,5 +16,5 @@ module.exports = {
     '^.+\\.(ts|js)$': 'ts-jest'
   },
 
-  silent: true
+  silent: false
```
Comment (Tadeuchi) [inline jest.config.js:19]: hmm?

File: src/__tests__/integration/internal-writes/internal-write-callee.test.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { WarpFactory } from '../../../core/WarpFactory';
 import { LoggerFactory } from '../../../logging/LoggerFactory';
 import { DeployPlugin } from 'warp-contracts-plugin-deploy';
 import { VM2Plugin } from 'warp-contracts-plugin-vm2';
+import { MemoryLevel } from "memory-level";
```
Comment (Tadeuchi) [inline src/__tests__/integration/internal-writes/internal-write-callee.test.ts:14]: lint?

File: src/core/modules/impl/DefaultStateEvaluator.ts [hunk 3]
```diff
@@ -155,11 +155,12 @@ export abstract class DefaultStateEvaluator implements StateEvaluator {
...
          Reading the state of the writing contract.
          This in turn will cause the state of THIS contract to be
-         updated in uncommitted state
+         updated in 'interaction state'
          */
         let newState: EvalStateResult<unknown> = null;
+        let writingContractState: SortKeyCacheResult<EvalStateResult<unknown>> = null;
         try {
-          await writingContract.readState(missingInteraction.sortKey, [
+          writingContractState = await writingContract.readState(missingInteraction.sortKey, [
```
Comment (Tadeuchi) [inline src/core/modules/impl/DefaultStateEvaluator.ts:163]: me like it

File: src/__tests__/integration/internal-writes/internal-write-callee.test.ts (hunk 2 above)
Comment (ppedziwiatr) [reply src/__tests__/integration/internal-writes/internal-write-callee.test.ts:14]: srint

**Pull request merged**